shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -782,8 +783,13 @@ public String getDartEntrypointFunctionName() {
    */
   @NonNull
   public String getInitialRoute() {
-    if (getIntent().hasExtra(EXTRA_INITIAL_ROUTE)) {
-      return getIntent().getStringExtra(EXTRA_INITIAL_ROUTE);
+    Intent intent = getIntent();
```

Comment:
how did you decide on the order? of whether to take the data string of the intent vs reading from the extra flag?

Reply:
if EXTRA_INITIAL_ROUTE is present, it means the users had very specifically instructed it. I would have it take precedence. 

String data in an intent could theoretically be anything and the engine doesn't do any validation here. Does the framework react in the right way if one just sends `adb shell am start --es "blah blah"`?

Reply:
--es will send the string to extra so it will not affect the intent.getData();

the getData always returns an URI and it will must passes the intent filter, so I think we should be fine